src/dist.h

```diff
@@ -0,0 +1,15 @@
+/*
+  Copyright (c) 2024, Intel Corporation
+
+  SPDX-License-Identifier: BSD-3-Clause
+*/
+
+/** @file dist.h
+    @brief Defines function with distribution specific implementations.
```

Comment:
It doesn't look for me as "distribution", it's about binary type.
I would rename this file to "binary_type.h".
Also I prefer separating build-related sources from compiler-core sources so I'd move dist.h, slim.h and whole.h to separate src/binary_type folder and rename them to binary_type.h, slim.h and composite.h. If you don't like idea of creating a subfolder it can be binary_type.h, binary_slim.h and binary_composite.h in src folder.

Reply:
Done.